test/specs/jsCodegen/nodeTests.tests.ts

```diff
@@ -91,4 +91,15 @@ describe('kind tests', () => {
 			documentNode.firstChild
 		);
 	});
+
+	it('handles detached nodes', () => {
```

Comment:
Do we need to go over the other paths-that-can-return-null as well? so child:: w/o children, attribute w/o matching attributes, etcetera?

Reply:
Yes. This is the cause for other errors.

Reply:
As long as you're not using firstchild/lastchild/sibling axes. :+1: